### ISSUE TITLE
Do not store wasm byte code in 08-wasm store

### ISSUE THREAD

**Issue by crodriguezvega** [cosmos/ibc-go#4085]
_Surfaced from Confio's review of 08-wasm at hash https://github.com/strangelove-ventures/ibc-go/commit/96e9930ab87a171a3ae6ac9717ee25bfda77212b_

See item 6.1 of report (TODO: add link when report is available in ibc-go). The wasm byte code is already stored by the wasm VM, so no need to store it in the IAVL store of 08-wasm. We will continue storing the code hashes of the contracts stored.

Pre-requisites:

- [x] #4091 
- [x]  #4185

Work items:
- Remove [`store.Set` operation in `storeWasmCode`](https://github.com/cosmos/ibc-go/blob/255905ae4a97d73cb4ef020fa9fedea9f943013a/modules/light-clients/08-wasm/keeper/keeper.go#L106).
- Replace the [`store.Get` in `Code` rpc handler](https://github.com/cosmos/ibc-go/blob/255905ae4a97d73cb4ef020fa9fedea9f943013a/modules/light-clients/08-wasm/keeper/grpc_query.go#L37) with `k.wasmVM.Getcode(codeHash)`.
- Change the [`KeyCodeIDPrefix`](https://github.com/cosmos/ibc-go/blob/2901cf8ff8ace220b709d009ebfebc4973c39881/modules/light-clients/08-wasm/types/keys.go#L15) to a key (so no prefix needed) under which we will store a list of code hashes (`codeHashes`).
- Add to `wasm.proto` a message like:
```
message CodeHashes {
  // list of code hashes
  repeated string code_hashes = 1;
}
```
- In `storeWasmCode` append any new code hash to the list and store it under the `codeHashes` key. If a contract that already has been stored is uploaded again, we can check that the code hash is already in the list and return early.
- Make the necessary changes in the [`CodeIds` gRPC handler](https://github.com/cosmos/ibc-go/blob/2901cf8ff8ace220b709d009ebfebc4973c39881/modules/light-clients/08-wasm/keeper/grpc_query.go#L48).

**Comment by damiannolan** [comment]
This removes storing the wasm byte code in the 08-wasm iavl, but didn't we speak about keeping an allow list of the codeHashes in 08-wasm?

If so, we can handle that in a separate PR outside of this issue, just wanted to raise the question here :)

**Comment by crodriguezvega** [comment]
Correct, this issue is only for removing the storing of the wasm code. We will still keep the code hashes in the store.

**Comment by crodriguezvega** [comment]
Ok, I changed the description of issue to handle also the storing of the code hashes. I think it makes sense to do it all together in this issue.

**Comment by damiannolan** [comment]
This entrypoint for `StoreCode` is gated by an `authority`, right?

This came up in the discussion last week iirc. If we are sharing the same vm instance between `x/wasm` and `08-wasm` then it should be valid to store client contract code via `x/wasm` permissionlessly, [for example on juno,](https://juno-api.w3coins.io/cosmwasm/wasm/v1/codes/params) but then it is not added to the allow list for instantiation? 

If you were to call `StoreCode` via `08-wasm` again with the same wasm code, is `wasmvm.StoreCode()` idempotent? or will it error? Should we maintain the allow list as a param and allow `authority` access to change it?
